programs/mmm/src/instructions/mpl_core_asset/mpl_core_wrap.rs

```diff
@@ -37,3 +51,66 @@ impl Deref for IndexableAsset {
         &self.0
     }
 }
+
+pub fn deserialize_collection_asset(
+    collection_account: &UncheckedAccount,
+) -> Result<Option<IndexableAsset>, Error> {
+    if collection_account.key == &Pubkey::default() {
+        return Ok(None);
+    }
+    let data = &collection_account.try_borrow_data()?;
```

Comment:
do we need assertion that the collection account is owned by the mpl core program?